libslide/src/bignum.rs

```diff
@@ -58,6 +70,9 @@ impl Bignum {
                 return Err(INPUT_ERR_MSG);
             }
         }
+        if check_zero(&dec) {
+            dec = vec![];
+        }
```

Comment:
Actually, you don't even need the check for whether the vector is empty in `is_zero`. Just check if it's all zeros (true if the vector is empty), and then call `clear` on the vector rather than allocating a new one.

Reply:
smart!